backend/app/fields.py

```diff
@@ -0,0 +1,27 @@
+"""
+Custom fields for the gender analysis web app.
+"""
+from django.db import models
+from django.utils.translation import gettext_lazy
+
+
+class LowercaseCharField(models.CharField):
+    """
+    A subclass of models.CharField that converts the string to lowercase
+    when saving the model instance to the database.
+    """
+
+    # WORK IN PROGRESS!
+    description = gettext_lazy(f"String (up to {super().max_length}) \
+            converted to lowercase upon saving the model instance to the database.")
+
+    def pre_save(self, model_instance, add):
```

Comment:
What is the purpose of the `add` parameter? It is not used in the method.

Reply:
> What is the purpose of the `add` parameter? It is not used in the method.

The `add` parameter is a Boolean attribute that specifies whether a Model instance should be added to the database or if Django should update a preexisting instance. This overridden `pre_save` method does not use the `add` parameter directly, but the superclass's `pre_save` method includes it and some other subclasses of `CharField` do use the `add` parameter (e.g., `DateField`). I'm looking at [these docs](https://docs.djangoproject.com/en/2.2/_modules/django/db/models/fields/) for reference.

While I do get a warning from PyCharm if I remove the unused `add` parameter, I'm not actually sure if removing it will cause a bug (the example I can think of is if Django sends a `pre_save` signal that calls the `pre_save` method of some unknown field passing in the `add` parameter--in that case, we'd need to keep it there?).

Reply:
That sounds good! Thanks for explaining :)